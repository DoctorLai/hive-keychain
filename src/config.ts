const Config = {
  hiveEngine: {
    mainNet: 'ssc-mainnet-hive',
  },
  claims: {
    FREQUENCY: +(process.env.DEV_CLAIM_FREQUENCY || 10 * 60 * 1000),
    freeAccount: {
      MIN_RC_PCT: +(process.env.DEV_CLAIM_ACCOUNT_RC_PCT || 95),
      MIN_RC: +(process.env.DEV_CLAIM_ACCOUNT_MIN_RC || 1.1 * 10 * 10 ** 12), // 10% more than 10^13 (current creation cost)
    },
  },
};

export default Config;
