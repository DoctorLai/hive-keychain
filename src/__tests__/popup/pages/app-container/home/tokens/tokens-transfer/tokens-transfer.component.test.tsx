import tokensTransfer from 'src/__tests__/popup/pages/app-container/home/tokens/tokens-transfer/mocks/tokens-transfer';
import alButton from 'src/__tests__/utils-for-testing/aria-labels/al-button';
import alComponent from 'src/__tests__/utils-for-testing/aria-labels/al-component';
import alIcon from 'src/__tests__/utils-for-testing/aria-labels/al-icon';
import alInput from 'src/__tests__/utils-for-testing/aria-labels/al-input';
import {
  EventType,
  QueryDOM,
} from 'src/__tests__/utils-for-testing/enums/enums';
import assertion from 'src/__tests__/utils-for-testing/preset/assertion';
import config from 'src/__tests__/utils-for-testing/setups/config';
import {
  clickAwait,
  clickTypeAwait,
} from 'src/__tests__/utils-for-testing/setups/events';
config.byDefault();
const { methods, constants, extraMocks } = tokensTransfer;
const { snapshotName, messages, selectedToken } = constants;
let _asFragment: () => {};
describe('tokens-transfer.component tests:\n', () => {
  methods.afterEach;
  describe('Snapshot cases', () => {
    beforeEach(async () => {
      _asFragment = await tokensTransfer.beforeEach();
    });
    it('Must show tokens balances with send icon', () => {
      expect(_asFragment()).toMatchSnapshot(
        snapshotName.tokens.with.transferIcons,
      );
    });
  });
  describe('With tokens balances', () => {
    beforeEach(async () => {
      await tokensTransfer.beforeEach();
      await clickAwait([alIcon.tokens.prefix.send + 'LEO']);
    });
    it('Must show error message if empty receiverUsername', async () => {
      await clickAwait([alButton.operation.tokens.transfer.send]);
      await assertion.awaitFor(messages.missingField, QueryDOM.BYTEXT);
    });
    it('Must show error message if 0 as amount', async () => {
      await clickAwait([alButton.operation.tokens.transfer.send]);
      await clickTypeAwait([
        {
          ariaLabel: alInput.username,
          event: EventType.TYPE,
          text: 'theghost1980',
        },
        {
          ariaLabel: alButton.operation.tokens.transfer.send,
          event: EventType.CLICK,
        },
      ]);
      await assertion.awaitFor(messages.missingField, QueryDOM.BYTEXT);
    });
    it('Must show error if negative amount', async () => {
      await clickAwait([alButton.operation.tokens.transfer.send]);
      await clickTypeAwait([
        {
          ariaLabel: alInput.username,
          event: EventType.TYPE,
          text: 'theghost1980',
        },
        {
          ariaLabel: alInput.amount,
          event: EventType.TYPE,
          text: '-1',
        },
        {
          ariaLabel: alButton.operation.tokens.transfer.send,
          event: EventType.CLICK,
        },
      ]);
      await assertion.awaitFor(messages.negativeAmount, QueryDOM.BYTEXT);
    });
    it('Must show error if unexistent user', async () => {
      extraMocks.doesAccountExist(false);
      await clickAwait([alButton.operation.tokens.transfer.send]);
      await clickTypeAwait([
        {
          ariaLabel: alInput.username,
          event: EventType.TYPE,
          text: 'theghost1980',
        },
        {
          ariaLabel: alInput.amount,
          event: EventType.TYPE,
          text: '1',
        },
        {
          ariaLabel: alButton.operation.tokens.transfer.send,
          event: EventType.CLICK,
        },
      ]);
      await assertion.awaitFor(messages.wrongUser, QueryDOM.BYTEXT);
    });
    it('Must show error if not enough balance', async () => {
      await clickAwait([alButton.operation.tokens.transfer.send]);
      await clickTypeAwait([
        {
          ariaLabel: alInput.username,
          event: EventType.TYPE,
          text: 'theghost1980',
        },
        {
          ariaLabel: alInput.amount,
          event: EventType.TYPE,
          text: (parseFloat(selectedToken.data.balance) + 1).toString(),
        },
        {
          ariaLabel: alButton.operation.tokens.transfer.send,
          event: EventType.CLICK,
        },
      ]);
      await assertion.awaitFor(messages.notEnoughBalance, QueryDOM.BYTEXT);
    });
    it.skip('Must show confirmation page and encrypted memo', async () => {
      //TODO check why the fill in teh fields is showing???
      extraMocks.doesAccountExist(true);
      extraMocks.getPublicMemo();
      await clickAwait([alButton.operation.tokens.transfer.send]);
      await clickTypeAwait([
        {
          ariaLabel: alInput.username,
          event: EventType.TYPE,
          text: 'cedricguillas',
        },
        {
          ariaLabel: alInput.amount,
          event: EventType.TYPE,
          text: '1',
        },
        {
          ariaLabel: alButton.operation.tokens.transfer.send,
          event: EventType.CLICK,
        },
      ]);
      await assertion.awaitFor(
        alComponent.confirmationPage + '0',
        QueryDOM.BYLABEL,
      );
    });
    it.todo('Must go back when cancelling transfer');
    it.todo('Must show popup_warning_phishing');
  });
  describe('No tokens balances', () => {});
  describe('No Active Key', () => {});
  describe('No Memo Key', () => {
    it.todo('Must show exhange validation warning');
  });
});
