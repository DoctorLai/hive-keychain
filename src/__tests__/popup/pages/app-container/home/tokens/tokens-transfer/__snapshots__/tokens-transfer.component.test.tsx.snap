// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokens-transfer.component tests:
 Snapshot cases:
 Having balances:
 Must show tokens balances with send icon: tokens-transfer.component ICONS 1`] = `
<DocumentFragment>
  <div
    class="App "
  >
    <div
      class="app-router"
      style="height: 100%; display: grid; grid-template-rows: 70px 1fr;"
    >
      <div
        class="title-section"
      >
        <span
          aria-label="arrow-back-icon"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          arrow_back
        </span>
        <div
          class="title"
        >
          Tokens
        </div>
        <span
          aria-label="icon-close-page"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          clear
        </span>
      </div>
      <div
        class="page-content"
        style="overflow: auto; display: flex; flex-direction: column;"
      >
        <div
          class="tokens-page"
        >
          <div
            class="disclaimer"
          >
            View your custom token balances. You can hide certain tokens and see more information by clicking the settings wheel.
          </div>
          <div
            class="top-bar-container"
          >
            <div
              class="custom-input"
            >
              <div
                class="input-container no-logo   "
              >
                <input
                  aria-label="input-filter-box"
                  class=""
                  placeholder="Search "
                  type="text"
                  value=""
                />
              </div>
            </div>
            <span
              aria-label="open-tokens-filter"
              class="icon-component material-icons-outlined filter clickable"
            >
              filter_alt
            </span>
            <span
              aria-label="tokens-settings-icon"
              class="icon-component material-icons-outlined settings clickable"
            >
              settings
            </span>
          </div>
          <div
            aria-label="my-tokens"
            class="my-tokens"
          >
            <div
              aria-label="token-user-item"
              class="token-item"
            >
              <div
                aria-label="token-user-symbol-PAL"
                class="token"
              >
                <div
                  class="tooltip-container"
                >
                  <div
                    class="tooltip-anchor"
                  >
                    <div
                      class="balance"
                    >
                      1,189.573
                    </div>
                  </div>
                </div>
                <div
                  class="symbol"
                >
                  PAL
                </div>
                <span
                  aria-label="icon-token-history-PAL"
                  class="icon-component material-icons-outlined history clickable"
                >
                  history
                </span>
                <span
                  aria-label="icon-send-history-PAL"
                  class="icon-component material-icons-outlined send clickable"
                >
                  send
                </span>
                <span
                  aria-label="icon-expand-more-PAL"
                  class="icon-component material-icons-outlined more closed clickable"
                >
                  expand_more
                </span>
              </div>
            </div>
            <div
              aria-label="token-user-item"
              class="token-item"
            >
              <div
                aria-label="token-user-symbol-BUILDTEAM"
                class="token"
              >
                <div
                  class="tooltip-container"
                >
                  <div
                    class="tooltip-anchor"
                  >
                    <div
                      class="balance"
                    >
                      100.000
                    </div>
                  </div>
                </div>
                <div
                  class="symbol"
                >
                  BUILDTEAM
                </div>
                <span
                  aria-label="icon-token-history-BUILDTEAM"
                  class="icon-component material-icons-outlined history clickable"
                >
                  history
                </span>
                <span
                  aria-label="icon-send-history-BUILDTEAM"
                  class="icon-component material-icons-outlined send clickable"
                >
                  send
                </span>
                <span
                  aria-label="icon-expand-more-BUILDTEAM"
                  class="icon-component material-icons-outlined more closed clickable"
                >
                  expand_more
                </span>
              </div>
            </div>
            <div
              aria-label="token-user-item"
              class="token-item"
            >
              <div
                aria-label="token-user-symbol-LEO"
                class="token"
              >
                <div
                  class="tooltip-container"
                >
                  <div
                    class="tooltip-anchor"
                  >
                    <div
                      class="balance"
                    >
                      38.861
                    </div>
                  </div>
                </div>
                <div
                  class="symbol"
                >
                  LEO
                </div>
                <span
                  aria-label="icon-token-history-LEO"
                  class="icon-component material-icons-outlined history clickable"
                >
                  history
                </span>
                <span
                  aria-label="icon-send-history-LEO"
                  class="icon-component material-icons-outlined send clickable"
                >
                  send
                </span>
                <span
                  aria-label="icon-expand-more-LEO"
                  class="icon-component material-icons-outlined more closed clickable"
                >
                  expand_more
                </span>
              </div>
              <div
                aria-label="token-info-expandable-panel"
                class="expandable-panel closed"
              >
                <div
                  class="token-info"
                >
                  <div
                    aria-label="token-info-go-to-website"
                    class="token-description"
                  >
                    <img
                      class="token-icon"
                      src="https://s3.amazonaws.com/steemmonsters/website/ui_elements/open_packs/img_essence-orb.png"
                    />
                    <div
                      class="token-name-issuer"
                    >
                      <span
                        class="token-name"
                      >
                        LEO tokens
                      </span>
                      <span
                        class="token-issuer"
                      >
                        @leofinance
                      </span>
                    </div>
                  </div>
                  <div>
                    Balance : 38.861
                  </div>
                  <div>
                    Staking : 1.060
                  </div>
                  <div
                    aria-label="button-go-to-incoming-delegations"
                    class="delegation-line"
                  >
                    Delegation in : 1
                    <span
                      class="icon-component material-icons-outlined  "
                    >
                      list
                    </span>
                  </div>
                  <div
                    aria-label="button-go-to-outgoing-delegations"
                    class="delegation-line"
                  >
                    <div>
                      Delegation out : 1
                    </div>
                    <span
                      class="icon-component material-icons-outlined  "
                    >
                      list
                    </span>
                  </div>
                </div>
                <div
                  class="button-panel"
                >
                  <div
                    aria-label="button-token-stake"
                    class="action-button stake"
                  >
                    Stake
                  </div>
                  <div
                    aria-label="button-token-unstake"
                    class="action-button unstake"
                  >
                    Unstake
                  </div>
                  <div
                    aria-label="button-token-delegate"
                    class="action-button delegate"
                  >
                    Delegate
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Toastify"
    />
    <div
      aria-label="proposal-voting-section"
      class="proposal-voting-section hide closed"
    >
      <span
        class="icon-component material-icons close clickable"
      >
        clear
      </span>
      <div
        class="text"
      >
        Support Keychain development proposal
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`tokens-transfer.component tests:
 Snapshot cases:
 No tokens balances:
 Must show no balances: tokens-transfer.component NO BALANCES 1`] = `
<DocumentFragment>
  <div
    class="App "
  >
    <div
      class="app-router"
      style="height: 100%; display: grid; grid-template-rows: 70px 1fr;"
    >
      <div
        class="title-section"
      >
        <span
          aria-label="arrow-back-icon"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          arrow_back
        </span>
        <div
          class="title"
        >
          Tokens
        </div>
        <span
          aria-label="icon-close-page"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          clear
        </span>
      </div>
      <div
        class="page-content"
        style="overflow: auto; display: flex; flex-direction: column;"
      >
        <div
          class="tokens-page"
        >
          <div
            class="disclaimer"
          >
            View your custom token balances. You can hide certain tokens and see more information by clicking the settings wheel.
          </div>
          <div
            class="top-bar-container"
          >
            <div
              class="custom-input"
            >
              <div
                class="input-container no-logo   "
              >
                <input
                  aria-label="input-filter-box"
                  class=""
                  placeholder="Search "
                  type="text"
                  value=""
                />
              </div>
            </div>
            <span
              aria-label="open-tokens-filter"
              class="icon-component material-icons-outlined filter clickable"
            >
              filter_alt
            </span>
            <span
              aria-label="tokens-settings-icon"
              class="icon-component material-icons-outlined settings clickable"
            >
              settings
            </span>
          </div>
          <div
            class="no-tokens"
          >
            You currently don't have any custom tokens. Click on the settings wheel to get information about the tokens available.
          </div>
        </div>
      </div>
    </div>
    <div
      class="Toastify"
    />
    <div
      aria-label="proposal-voting-section"
      class="proposal-voting-section hide closed"
    >
      <span
        class="icon-component material-icons close clickable"
      >
        clear
      </span>
      <div
        class="text"
      >
        Support Keychain development proposal
      </div>
    </div>
  </div>
</DocumentFragment>
`;
